### ISSUE TITLE
Predefined regex

### ISSUE THREAD

**Issue by adamwojszczyk** [Codearte/accurest#115]
I would like to be able to use some predefined regex email, host, etc.

Right know I handle it by creating variables to make it more readable but it does not scale.
Solution should allow to concatenate them with custom values.

For url it may look like this:

``` groovy
headers {
            header 'Location': $(client('http://partners.com/partners/1000/users/1001'), server(regex("${host}/partners/1000/users/1001")))
        } 
```

Host regex should match localhost and real hosts.

**Comment by marcingrzejszczak** [comment]
I would create such a class:

```
@CompileStatic
class RegexPatterns {

    private static final Pattern IP_ADDRESS = Pattern.compile('([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])'); 
    private static final Pattern HOSTNAME_PATTERN = Pattern.compile('((http[s]?|ftp):\\/)\\/?([^:\\/\\s]+)(:[0-9]{1,5})?');
    private static final Pattern EMAIL = Pattern.compile('[_]*([a-z0-9]+(\\.|_*)?)+@([a-z][a-z0-9-]+(\\.|-*\\.))+[a-z]{2,6}');
    private static final Pattern URL = Pattern.compile('((www\\.|(http|https|ftp|news|file)+\\:\\/\\/)[_.a-z0-9-]+\\.[a-z0-9\\/_:@=.+?,##%&~-]*[^.|\\\'|\\# |!|\\(|?|,| |>|<|;|\\)])');

    String ipAddress() {
        return IP_ADDRESS.pattern()
    }

    String hostname() {
        return HOSTNAME_PATTERN.pattern()
    }

    String email() {
        return EMAIL.pattern()
    }

    String url() {
        return URL.pattern()
    }
}

```

it would give you a possibility to write sth like this:

```
headers {
            header 'Location': $(client('http://partners.com/partners/1000/users/1001'), server(regex("${host()}/partners/1000/users/1001")))
        } 
```
- would that be ok?
- anything else to add there for regex?

**Comment by marcingrzejszczak** [comment]
@adam-wojszczyk - would that be ok  https://github.com/Codearte/accurest/commit/e0436ace9fd62744b12b17994dc51eb9e455537d#diff-ff0c31096d76fe449d5a32fee665dd7eR366 ?

Here are the regexps:
https://github.com/Codearte/accurest/commit/e0436ace9fd62744b12b17994dc51eb9e455537d#diff-ff7febbe63611417373e094f588b83baR8

**Comment by OlgaMaciaszek** [comment]
Implemented in 0.8.3.
